doc/dev/autograph.rst

```diff
@@ -214,29 +215,19 @@ Instead of utilizing a return statement, use the following approach instead:
             y = x ** 3
         return y
 
-Different branches must assign the same type
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Automatic type promotion in branches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think this section is still useful to point out that you must return the same "type" of result, where type is more generally interpreted to mean the same PyTree and the same array shapes, not necessarily just the same data type (`dtype`).

Reply:
Oh good point